client/src/pages/LoginDialog.js

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { Button, Collapse, Dialog, DialogActions, DialogContent, DialogTitle, Divider, 
+  InputLabel, Link, OutlinedInput, Typography } from '@material-ui/core';
+import { Alert } from '@material-ui/lab';
+
+function LoginDialog() {
+  const [email, setEmail] = React.useState("");
+  const [password, setPassword] = React.useState("");
+  const [emailErrorOpen, setEmailErrorOpen] = React.useState(false);
+  const [passwordErrorOpen, setPasswordErrorOpen] = React.useState(false);
+  
+  const handleClose = () => {
+    window.location.href = "/";
+  };
+  
+  const openSignUp = () => {
+    window.location.href = "/sign-up";
+  };
+
+  const changeEmail = event => {
+    setEmail(event.target.value);
+  };
+
+  const changePassword = event => {
+    setPassword(event.target.value);
+  };
+  
+  const validateEmail = email => {
+    let regExpression = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    return regExpression.test(String(email).toLowerCase());
+  };
+  
+  const validatePassword = password => {
+    return password.length >= 6;
+  };
+  
+  const validateLogin = () => {
+    let emailValidated = validateEmail(email);
+    let passwordValidated = validatePassword(password);
+    if (!emailValidated) {
+      setEmailErrorOpen(true);
+    } if (emailValidated) {
+      setEmailErrorOpen(false);
+    } if (!passwordValidated) {
+      setPasswordErrorOpen(true);
+    } if (passwordValidated) {
+      setPasswordErrorOpen(false);
+    } if (emailValidated && passwordValidated) {
+      localStorage.setItem('email', email);
+      localStorage.setItem('password', password);
```

Comment:
Don't store the password in local storage. If a user enters a malicious website, all it would take is a localStorage.getItem("password") to get the password.

Reply:
We also might want to look into react context api for global state opposed to localStorage.